File: problems/1975_maximum_matrix_sum/1975_maximum_matrix_sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        total_sum = 0
        min_abs_val = float('inf')
        neg_count = 0

        for row in matrix:
            for val in row:
                if val < 0:
                    neg_count += 1
                    val = -val
                total_sum += val
                min_abs_val = min(min_abs_val, val)
        
        if neg_count % 2 != 0:
            total_sum -= 2 * min_abs_val
        
        return total_sum

File: problems/1975_maximum_matrix_sum/1975_maximum_matrix_sum.java
class Solution {
    public long maxMatrixSum(int[][] matrix) {
        long totalSum = 0;
        int minAbsVal = Integer.MAX_VALUE;
        int negCount = 0;

        for (int i = 0; i < matrix.length; i++) {
            for (int j = 0; j < matrix[0].length; j++) {
                int val = matrix[i][j];
                if (val < 0) {
                    negCount++;
                    val = -val; // Do abs
                }
                totalSum += val;
                minAbsVal = Math.min(minAbsVal, val);
            }
        }

        if (negCount % 2 != 0) {
            totalSum -= 2 * minAbsVal;
        }

        return totalSum;
    }
}